src/lib/tls/asio/asio_error.h

```diff
@@ -56,6 +92,7 @@ struct BotanErrorCategory : boost::system::error_category
    {
    const char* name() const noexcept override
       {
+      // TODO: this seems wrong given the method brief
```

Comment:
Strictly speaking, this string is exposed to the outside and could be used by someone to check what kind of error occurred. But I think we can still change it, right? Maybe we should refine all the names. `StreamCategory`'s name looks inconsistent as well.

Reply:
Addressed in the comment below.